src/vgg.jl

```diff
@@ -144,117 +145,38 @@ backbone(m::VGG) = m.layers[1]
 classifier(m::VGG) = m.layers[2]
 
 """
-    VGG11(; pretrain = false, batchnorm = false)
+    VGG(depth::Int = 16; pretrain = false, batchnorm = false)
 
-Create a VGG-11 style model
+Create a VGG style model with specified `depth`. Available values include (11, 13, 16, 19).
 ([reference](https://arxiv.org/abs/1409.1556v6)).
 See also [`VGG`](#).
 
 !!! warning
-    `VGG11` does not currently support pretrained weights.
+    `VGG` does not currently support pretrained weights.
 
 # Arguments
 - `pretrain`: set to `true` to load pre-trained model weights for ImageNet
 """
-function VGG11(; pretrain = false, batchnorm = false)
-  model = VGG((224, 224); config = vgg_config[:A],
-                          inchannels = 3,
-                          batchnorm = batchnorm,
-                          nclasses = 1000,
-                          fcsize = 4096,
-                          dropout = 0.5)
-
-  if pretrain && !batchnorm
-    loadpretrain!(model, "VGG11")
-  elseif pretrain
-    loadpretrain!(model, "VGG11-BN)")
-  end
-  return model
-end
-
-"""
-    VGG13(; pretrain = false, batchnorm = false)
-
-Create a VGG-11 style model
-([reference](https://arxiv.org/abs/1409.1556v6)).
-See also [`VGG`](#).
-
-!!! warning
-    `VGG13` does not currently support pretrained weights.
-
-# Arguments
-- `pretrain`: set to `true` to load pre-trained model weights for ImageNet
-"""
-function VGG13(; pretrain = false, batchnorm = false)
-  model = VGG((224, 224); config = vgg_config[:B],
-                          inchannels = 3,
-                          batchnorm = batchnorm,
-                          nclasses = 1000,
-                          fcsize = 4096,
-                          dropout = 0.5)
-
-  if pretrain && !batchnorm
-    loadpretrain!(model, "VGG13")
-  elseif pretrain
-    loadpretrain!(model, "VGG13-BN)")
-  end
-  return model
-end
-
-"""
-    VGG16(; pretrain = false, batchnorm = false)
-
-Create a VGG-11 style model
-([reference](https://arxiv.org/abs/1409.1556v6)).
-See also [`VGG`](#).
-
-!!! warning
-    `VGG16` does not currently support pretrained weights.
+function VGG(depth::Int = 16; pretrain = false, batchnorm = false)
+  @assert depth in keys(vgg_config) "depth must be from one in $(sort(collect(keys(vgg_config))))"
 
-# Arguments
-- `pretrain`: set to `true` to load pre-trained model weights for ImageNet
-"""
-function VGG16(; pretrain = false, batchnorm = false)
-  model = VGG((224, 224); config = vgg_config[:D],
+  model = VGG((224, 224); config = vgg_conv_config[vgg_config[depth]],
```

Comment:
Any reason for the double dict?

Reply:
Mostly to mirror the other models and to retain the same nomenclature of the model config as the paper. I was considering changing the config such that we go `11 => A => ...` too.